### PR TITLE
Add solhint

### PR DESCRIPTION
**Pull request by Agupane**
## Description

-Replaces the old `solium` dependency as it's deprecated in flavor of [solhint](https://github.com/protofire/solhint) for linting the solidity code
-I used the recommended config rules and disabled all styling rules (and some others, like no-inline-assembly)

This is the output right now:
```
contracts/test-helpers/TokenExchanger.sol
  19:5  warning  Open bracket must be on same line. It must be indented by other constructions by space  bracket-align

✖ 1 problem (0 errors, 1 warning)

```

#### What is being addressed by this PR?
Add a linter to the project

#### What is the motivation for this change?
Replace a deprecated dependency

#### How is the change going to be implemented?
-Adding the solhint package to the package.json file.
-Adding a solhint config to the project.

#### What packages that will be affected by this change?
None
#### What PRs will be affected by this change?
None
#### What is not addressed by this PR?
Fixes for warnings and errors found by the linter

### Note:
If you wanna add styling rules, you can add the prettier [plugin](https://github.com/fvictorio/solhint-plugin-prettier) for solhint

Regards

### CODE REVIEW

**Comment by justinjmoses** [comment]
Thanks for the PR @Agupane - definitely a task that needed doing! We're just going to look over this later today and get back to you.

File: .solhint.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "no-inline-assembly": "off",
+    "no-simple-event-func-name": "off",
+    "separate-by-one-line-in-contract": "off",
+    "two-lines-top-level-separator": "off",
+    "var-name-mixedcase": "off",
+    "visibility-modifier-order": "off",
+    "mark-callable-contracts": "off",
+    "compiler-version": ["error", "0.4.25"]
+  }
+}
```
Comment (justinjmoses) [inline .solhint.json]: [np] would you mind adding a newline here?

Comment (Agupane) [reply .solhint.json]: Sure :), it's done now

File: package.json [hunk 2]
```diff
@@ -87,6 +86,7 @@
 		"commander": "^2.19.0",
 		"openzeppelin-solidity": "2.0.0",
 		"pretty-error": "^2.1.1",
-		"solc": "0.4.25"
+		"solc": "0.4.25",
+		"solhint": "^2.3.0"
```
Comment (justinjmoses) [inline package.json]: This should be a dev-dependency. But I'll fix it in our community branch.

**Pull request merged**